test/test-ncu.js

```diff
@@ -81,31 +80,34 @@ describe('npm-check-updates', function () {
                 upgradeAll: false
             });
 
-            return upgraded.should.eventually.not.have.property('escape-string-regexp')
-        })
+            return upgraded.should.eventually.not.have.property('escape-string-regexp');
+        });
     });
 
-    describe('cli', function() {
+    describe('cli', function () {
 
-        it('should accept stdin', function() {
+        it('should accept stdin', function () {
             return spawn('node', ['bin/ncu'], '{ "dependencies": { "express": "1" } }')
                 .then(function (output) {
-                    output.trim().should.startWith('express')
+                    output.trim().should.startWith('express');
                 });
         });
 
-        it('should fall back to package.json search when receiving empty content on stdin', function (done) {
-            var childProcess = child.exec('node bin/ncu', function (error, stdout) {
-                if (error) {
-                    done(error);
-                }
+        it('should fall back to package.json search when receiving empty content on stdin', function () {
+            return spawn('node', ['bin/ncu']).then(function (stdout) {
                 stdout.toString().trim().should.match(/^Using .+package.json/);
-                done();
             });
-            childProcess.stdin.end();
         });
 
-        it('should output json with --jsonAll', function() {
+        it('should handle no package.json to analyze when receiving empty content on stdin', function () {
+            // run from tmp dir to avoid ncu analyzing the project's package.json
+            return spawn('node', [process.cwd() + '/bin/ncu'], {cwd: tmp.dirSync().name})
```

Comment:
Changing the cwd of the child process to a tmp dir will make it look for 'bin/ncu' in the tmp dir, I'm using the parent process cwd for an abs path